src/coord/src/catalog.rs

```diff
@@ -435,6 +436,8 @@ impl CatalogState {
             CatalogItem::Table(_) => Volatile,
             CatalogItem::Type(_) => Unknown,
             CatalogItem::Func(_) => Unknown,
+            // TODO: what does volatile mean
+            CatalogItem::Secret(_) => Volatile,
```

Comment:
needs PR attention

Reply:
It's an outdated term for "not definite". See: https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/overview/volatility.md

I'd go with `Nonvolatile`, so that a source that depends on a secret doesn't get marked as "volatile". But it doesn't much matter, because we'll probably swap this for "definite" before too long.